### DISCUSSION TITLE
Como começar?

### DISCUSSION THREAD

**Discussion by HukBross257** [frontendbr/forum#2455] [Perguntas e Respostas]
``*Eu estou com dúvida de como começar a ser um programador. Se você puder me ajudar com dicas eu agradeceria bastante.*``

**Comment by jampow** [comment]
Acho q um bom meio, quando se está perdido, seria fazer um curso. Tem vários gratuitos no YouTube, ou se estiver disposto a pagar, eu sugeriria a alura. Acho q pode te dar uma noção melhor de boa parte do q é possível ser feito com programação e pode ajudar a definir um caminho a seguir.

**Reply by HukBross257** [reply]
a sim, obrigado por me ajudar estava precisando de dicas sobre.

**Reply by jampow** [reply]
tbm tem este projeto open source que pode tentar usar como guia.

https://roadmap.sh/

**Reply by HukBross257** [reply]
boa

**Comment by OtavioXimarelli** [comment]
Cara eu também tô começando nesse mundo,  o que está me ajudando é a plataforma freecodecamp.org, eles tem um material bem legal lá, com vários projetos práticos, da uma olhada lá.
Para você ter uma ideia dos conteúdos, esse foi o site que eles pedem para criar já na introdução do curso de desenvolvimento web: 👇🏼 https://form-otavio-beta.vercel.app/

**Reply by OtavioXimarelli** [reply]
Pode sim irmão!

**Comment by ghost** [comment]


**Reply by HukBross257** [reply]
Pode ser

**Comment by AlexsandroSA** [comment]
### Meus 10 centavos 😄

> _Vou assumir que decidiu focar em frontend então tudo que comentar é com base nessa hipótese _ 

Focar na base e pratica/aplica todo conhecimento adquirido.  Fiz isso no inicio e não me arrependo.

**Foco na base**
 - HTML e CSS -> https://www.youtube.com/watch?v=Ejkb_YpuHWs&list=PLHz_AreHm4dkZ9-atkcmcBaMZdmLHft8n
 - Javascript -> https://www.youtube.com/watch?v=BXqUH86F-kA&list=PLntvgXM11X6pi7mW0O4ZmfUI1xDSIbmTm

Quando se sentir confortável com sugiro e estudar algum framework ou lib e cria alguns projetos com os mais famosos (react, angular, VueJs, Svelte...etc) e ver o que tu mais curte.
 
**Praticar/aplicar conhecimentos adquiridos**

 - [Desafios diários de programação](https://github.com/AlexsandroSA/daily-programming-challenges) (pequeno jaba - meu repo)

Nesse repo tu vai encontrar alguns sites para exercitar a base que comentei antes, se ingles não for um problema o [Javascript 30](https://javascript30.com/) é um bom começo.

**Reply by Scrip123** [reply]
Eu sujeria começando com Linux primeiro... O básico, ou seja, um curso de linux LPI, tem alguns no youtube, e depois aprender um pouco de linguagem C. E não aprender já de cara HTML e CSS.

Aprendendendo o básico de linux, ele terá facilidade em manejar o S.O, assim sendo, terá facilidade de aprender uma linguagem de programação

**Reply by AlexsandroSA** [reply]
Hum... bom ponto @Scrip123 de fato estudar/entender um SO ajuda _(em especial no processo de aprender, algo mais a médio longo prazo)_

Acabei "pulando" SO e focando apenas em frontend no curto prazo.

Grato por compartilhar seu pontos de vista :D

**Comment by CassiusHorvath** [comment]
**### Uma historinha para quem está a começar**

Se me permitem vou contar um bocadinho da minha história, espero que ajude a motivar quem está começando. Texto longo mas acho que vale a pena a leitura.

Eu atualmente tenho 51 anos, me formei em Bacharelado em Informática em 2000. Acabei por não trabalhar como engenheiro de sistemas e cai para a área de hardware, redes e administração de sistemas. Não reclamo, fiz minha vida no Brasil (spoiler) com isso.
Independente da veia política de cada um, em 2012 vi que a economia e as coisas estavam começando a ir para um caminho esquisito. Em 2013, dois amigos deixaram o Brasil para tentar uma vida melhor em outros cantos, um era piloto de Airbus e foi para a China e outro engenheiro civil e foi para Nova Zelândia.
Presenciei várias situações de violência urbana, minha filha foi assaltada duas vezes indo para a pós-graduação e sofreu uma tentativa de sequestro que felizmente conseguiu se desvencilhar dos meliantes. Enfim, um horror, sensação de ser prisioneiro na sua própria casa. Tudo isso, junto com a nóticia dos meus amigos indo embora, foi o gatilho que me faltava para tomar a decisão de ir embora do Brasil também.
Obviamente estudei o mercado de trabalho, e vi que para conseguir emprego fora seria infinitamente mais fácil como programador, ou em "fancy mode" engenheiro de software. Estava a quase 15 anos afastado desse mundo e assim como vocês que estão começando agora eu me questionei ... e agora? Para onde correr?

Tinha uma situação financeira favorável e comprei um curso presencial, muito bom por sinal, me deu uma base muito interessante de Java e uma fantástica volta ao mundo dos banco de dados (modulo de banco de dados foi muito bom mesmo). Parte de Frontend foi bem básica, beirando ao mediocre. 
Mas foram 6 meses que me ajudaram a perceber o abismo da falta de conhecimento que tinha que superar. Tentei em vão conseguir um estagio para ganhar experiência, até propus trabalhar de graça, só queria ganhar vivência no mundo real do desenvolvimento de software. Nada, acreditem.. nem de graça.. kkkkk
Parti então para fazer cursos pela Udemy, como minha base de frontend era fraca foi exatamente por ai que comecei. Fiz o curso de frontend do Daniel Tapias Morales que me deu uma excelente base de HTML, CSS e javascript. Depois fiz o curso de Angular do Maximilian Schwarzmüller. 

De 2014 a 2016 foram cursos e uma busca incansável por algum trabalho fora do Brasil que me desse oportunidade de estar legalmente no país, no caso estava tentando Canadá, Irlanda, Austrália, Nova Zelândia e Portugal. Canadá e Irlanda foram os dois primeiros que desisti por motivos que posso contar depois. A Austrália também virou carta fora do baralho pois o processo de imigração era mais rígido e complicado que o da Nova Zelândia. Ficou então NZ e Portugal ainda na mesa, mas nada de nada. Minha esposa não queria ir para a NZ por ser muito longe, etc, eu não me importava. 

Resumindo, em 2016 consegui uma oferta de emprego em Portugal, não como programador, mas dentro da area de hardware que é onde eu tinha mais experiencia. Salário não era bom mas eu queria poder chutar tudo e depois buscaria algo melhor. Continuei a estudar e me aperfeiçoar na area de desenvolvimento de software, Git, conceitos de infraestrutura, REST, SOLID, tudo que podia acrescentar alguma coisa. Obviamente me faltava experiência de já ter trabalhado em algum canto, mas tinha fé que apesar de ja estar com meus 43 anos (naquela data) iria conseguir me realocar no mercado de trabalho em Portugal.

Essa foi minha vida até 2018, quando finalmente abriu a primeira porta!!!! Consegui finalmente uma vaga de engenheiro de software em uma consultura francesa aqui na cidade do Porto. Salário era muito melhor do que ganhava, me deram carro da empresa, e outras regalias. Finalmente ia conseguir ter alguma experiência real!!!

Me dediquei, subi a lider de equipe da parte de frontend e fiquei nessa empresa por quase um ano e meio quando a segunda porta se abriu :). Ainda estou nessa porta que mudou minha vida. Hoje sou Senior Software Engineer na BMW. Já passei por vários projetos que me deram incontáveis experiências, viajo regularmente para alemanha, e sou muito muito feliz com a vida que tenho hoje.  Espero que essa minha história inspire aqueles que estão começando agora e aqueles que estão com dificultades que.... saibam, uma hora a porta abre!!

- dedicação
- foco
- fuja da zona de conforto
- busque incansavelmente por seus sonhos, só você pode realizá-los!!!

Uma última coisa independente se ficam no Brasil ou querem ir para fora, aprendam PELO MENOS inglês, é essencial!!!!

Obrigado por lerem!

**Comment by Emanoellima-dev** [comment]
No início é bem complicado mesmo decidir pra qual rumo seguir. Tbm tô meio perdido!!

**Reply by maxxcleiton** [reply]
Começa em lógica de programação com html e css, e vê sobre linux e servidores, talvez curta mais o lado infraestrutura do que códigos.

**Comment by felipefialho** [comment]
Uns bons anos atrás escrevi esse artigo que vai muito de encontro sobre esse tema: [Como estudar, praticar e conseguir oportunidades como (Front-end) Developer](https://www.felipefialho.com/blog/como-estudar-praticar-e-conseguir-oportunidades-como-front-end-developer/), já é old mas ainda é (g)old 

Vale dar uma lida 🚀

**Comment by ghost** [comment]
Eu geralmente indico aprender uma tecnologia e uma área de conhecimento: vuejs para frontend, react-native para mobile e afins. Ou por outra tenta pensar numa carreira, qual lugar que você quer, pode ou deve trabalhar. Quer trabalhar em pequena empresa? média empresa? grande empresa? No Google? No Facebook? Numa empresa governamental?  Acha melhor fazer concurso público ou privado? Quer trabalhar por conta própria? Quer ter o próprio negócio?

**Edit:** Removi a resposta anterior por achar que era uma resposta minimalista e simples.
